src/test/rustdoc-js-std/parser-errors.js

```diff
@@ -0,0 +1,295 @@
+const QUERY = [
+    '<P>',
+    '-> <P>',
+    'a<"P">',
+    '"P" "P"',
+    'P "P"',
+    '"p" p',
+    '"const": p',
+    "a<:a>",
+    "a<::a>",
+    "((a))",
+    "(p -> p",
+    "::a::b",
+    "a::::b",
+    "a::b::",
+    ":a",
+    "a b:",
+    "a (b:",
+    "_:",
+    "a-bb",
+    "a>bb",
+    "ab'",
+    "a->",
+    '"p" <a>',
+    '"p" a<a>',
+    "a,<",
+    "aaaaa<>b",
+    "fn:aaaaa<>b",
+    "->a<>b",
+    "a<->",
```

Comment:
A couple cases where rustdoc-search-ref used to disagree with search.js and also with the eBNF. Worth adding to this list of weird corner cases:

    a:: a
    a ::a

It's a parser error, and it probably should be a parser error, but it depends on how you handle spaces, which is full of them.

Reply:
I added tests to confirm the parser error. Also, in which case could they be valid? A path cannot start or end with `::` from the eBNF:

```
path = ident *(DOUBLE-COLON ident)
```

Did I miss something?

Reply:
You didn’t miss anything. I did. The parser was using a tokenizer that ignored spaces around the `::` like how Rust’s parser does.